packages/react/src/components/Pagination/index.tsx

```diff
@@ -0,0 +1,157 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import IconButton from '../IconButton';
+import TooltipTabstop from '../TooltipTabstop';
+import Icon from '../Icon';
+
+type PaginationVariant = 'dark';
+
+interface Props {
+  totalItems: number;
+  itemsPerPage?: number;
+  currentPage?: number;
+  statusLabel?: React.ReactElement;
+  firstPageLabel?: string;
+  previousPageLabel?: string;
+  nextPageLabel?: string;
+  lastPageLabel?: string;
+  onNextPageClick?: () => void;
+  onPreviousPageClick?: () => void;
+  onFirstPageClick?: () => void;
+  onLastPageClick?: () => void;
+  className?: string;
+  variant?: PaginationVariant;
+}
+
+const Pagination = React.forwardRef<HTMLDivElement, Props>(
+  (
+    {
+      totalItems,
+      itemsPerPage = 10,
+      currentPage = 1,
+      statusLabel,
+      firstPageLabel = 'First page',
+      previousPageLabel = 'Previous page',
+      nextPageLabel = 'Next page',
+      lastPageLabel = 'Last page',
+      onNextPageClick,
+      onPreviousPageClick,
+      onFirstPageClick,
+      onLastPageClick,
+      className,
+      variant,
+      ...other
+    },
+    ref
+  ) => {
+    const itemStart = currentPage * itemsPerPage - itemsPerPage + 1;
+    const itemEnd = Math.min(itemStart + itemsPerPage - 1, totalItems);
+    const isLastPage = itemEnd === totalItems;
+
+    return (
+      <div
+        ref={ref}
+        className={classNames('Pagination', className, {
+          'Pagination--dark': variant === 'dark'
+        })}
+        {...other}
+      >
+        <ul>
+          <li>
+            {currentPage === 1 ? (
+              <TooltipTabstop tooltip={firstPageLabel} placement="bottom">
```

Comment:
Default placement seems odd, I guess we're always assuming the pagination is on the bottom of the thing its paginating?

Reply:
I agree. I made it bottom because I assume pagination will primarily be below a table. The default was unacceptable because it covered up other buttons in the pagination component.  I'm gathering that we should probably make this configurable via a prop (defaulting to `bottom` still makes sense to me 🤷 )